kaiax/gasless/impl/init.go

```diff
@@ -45,25 +48,28 @@ func NewGaslessModule() *GaslessModule {
 	return &GaslessModule{}
 }
 
-func (g *GaslessModule) Init(opts *InitOpts) error {
-	if opts == nil || opts.ChainConfig == nil || opts.NodeKey == nil || opts.TxPool == nil {
-		return ErrInitUnexpectedNil
+func (g *GaslessModule) Init(opts *InitOpts) (disabled bool, err error) {
+	if opts == nil || opts.ChainConfig == nil || opts.NodeKey == nil || opts.Chain == nil || opts.TxPool == nil {
+		return true, ErrInitUnexpectedNil
 	}
-	g.InitOpts = *opts
 
-	g.swapRouters = map[common.Address]bool{
-		common.HexToAddress("0x1234"): true,
-	}
-	g.allowedTokens = map[common.Address]bool{
-		common.HexToAddress("0xabcd"): true,
+	if opts.ChainConfig.Gasless == nil || opts.ChainConfig.Gasless.IsDisabled {
+		return true, nil
 	}
+
+	g.InitOpts = *opts
 	g.signer = types.LatestSignerForChainID(g.ChainConfig.ChainID)
-	return nil
-}
 
-func (g *GaslessModule) Start() error {
-	return nil
-}
+	backend := backends.NewBlockchainContractBackend(g.Chain, nil, nil)
+	g.swapRouter, err = system.ReadActiveAddressFromRegistry(backend, GaslessSwapRouterName, g.Chain.CurrentBlock().Number())
+	if err != nil {
+		return true, err
```

Comment:
I'm not sure, if an error is returned in L66 and L71, it means that the preparation of the contract required for gasless tx has failed. Is it okay to return true?

Reply:
You mean this should panic?

Reply:
Yes, but after thinking about it again, I thought no problem because err is returned in backend.go.